src/main/java/org/example/daos/JobRoleDao.java

```diff
@@ -0,0 +1,58 @@
+package org.example.daos;
+
+import org.example.exception.DatabaseConnectionException;
+import org.example.models.JobRole;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JobRoleDao {
```

Comment:
This class should really be an interface with a separate concrete implementation specific to SQL so we're following some clean architecture principles. If we were to do this, we are decoupling the service that calls this method from the details of the data storage and making our code more modular. We want our code to be modular, it won't happen in this project but say we change from a sql db to no sql. We just need to write a new concrete implementation and then the code in the service doesn't need to change.

Link here that looks to explain the pattern: https://www.oracle.com/java/technologies/data-access-object.html

Let me know what you think :)

Reply:
Yes I think that's a more appropriate design method, went ahead and creating the interface!